### PR TITLE
Convert to an always-on F1 instance (and other related changes)

### PR DESCRIPTION
**Pull request by redfern314**
## Changes
* Convert to a smaller F1 instance which is always-on
* Improve bootstrap documentation for SFNF contributors
* Update Node.js to v18 (v16 is EOL)

## Testing
* Deployed this branch to staging version (see additions to README); observed that app worked normally, though instance did not stay on for more than 15min (this appears to be expected, traffic must be routed to the version in order for min_instances to take effect)
* Routed traffic to staging version and observed that instance stayed on for over an hour (presumably indefinitely)
* Checked that hammering the site from multiple browser tabs didn't explode anything
* Reverted traffic to previous mainline version while I wait for approval on this PR

## Billing
I intend to let this run for a week or so, then check back to see what effect it has on our bill. Google Cloud doesn't have good forecasting so I think the best way to check how much we're being charged is to just... try it out. (It should be comparable to what we're already paying, though, which is not much.)

## Caveat...
The downside to this change is that people hammering the chive on Tuesdays will spin up extra instances since F1 is smaller. Hopefully not too many, though.

### CODE REVIEW

File: app.yaml
Comment (eplusminus) [inline app.yaml:8]: I really appreciate all the comments you added, but I especially love it when people include URLs to the source of the relevant information like you did here.

**Review by eplusminus — APPROVED** [review]
I still know almost nothing about Google Cloud, but your explanation of these changes made sense to me and if it's working in staging, go for it!

File: app.yaml
Comment (redfern314) [reply app.yaml:8]: It's as much for me as for anyone else! I'm notoriously sieve-minded if I don't have solid notes in place to reproduce something I figured out months ago. Bites me at work sometimes.

**Comment by redfern314** [comment]
Thank you! I'll wait to see if Topher has thoughts and then apply it :)

File: server/index.js
Comment (eplusminus) [inline server/index.js:43]: Trivial note: it looks to me from the docs like Google Cloud doesn't care what the response is from this handler; the call is just for the application's benefit (that's different from the health check request, where Google Cloud can be configured to expect a specific string like "OK" in the response). And since the default in Express is to send a 200 status with an empty response body ([a behavior](https://nodejs.org/en/learn/modules/anatomy-of-an-http-transaction#http-status-code) it inherits from the underlying Node response class), you could omit this `res.send` call entirely.

**Comment by clizzin** [comment]
Looked this over, and while I don't have experience with these aspects of Google Cloud, it all seems good! Thank you for doing this! Really helpful to learn how to deploy a test version too.

I suggest deploying sometime other than Tuesday or Friday evening, otherwise I leave it to you to decide how to roll it out.

**Comment by clizzin** [comment]
I just took a look at the error reporting tool in Google Cloud, and this happened in staging, but I'm not sure if it's of concern. Did you already investigate this?
https://console.cloud.google.com/errors/detail/CMHlur_a3_PsngE?project=sfneofuturists-chive

File: server/index.js
Comment (redfern314) [reply server/index.js:43]: That's fair. Doing this actually did help me during dev since it's simple to tell that adding the route worked, but it's not needed for prod. I'll remove it before merging.

**Comment by redfern314** [comment]
@clizzin that error was while I was still trying to deploy a working version; I was playing around with APPROVED_DOMAINS and gave it something invalid. Good catch, though.

File: server/index.js
Comment (redfern314) [reply server/index.js:43]: Actually, omitting it appears to cause the connection to hang client-side. We could use `res.sendStatus(200)` instead, but I'd rather not just let it hang in case that makes Google's side act weird. Going to leave this in.

**Comment by redfern314** [comment]
Merging the PR now; I'll deploy it to the site on Weds.

**Pull request merged**
